src/Symfony/Component/Form/Util/ParamFilesMerger.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Util;
+
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * @author Priyadi Iman Nurcahyo <priyadi@rekalogika.com>
+ */
+class ParamFilesMerger
+{
+    private array $path;
+    private array $params;
+    private array $files;
+
+    /**
+     * @param array $path   The path to the current element, empty means the root
+     * @param array $params The parameters
+     * @param array $files  The files
+     */
+    public function __construct(array $path, array $params, array $files)
+    {
+        $this->path = $path;
+        $this->params = $params;
+        $this->files = $files;
+    }
+
+    public function getResult()
+    {
+        $paramsValue = $this->getParamsValue();
+        $filesValue = $this->getFilesValue();
+
+        if (null === $paramsValue) {
+            if (null === $filesValue) {
+                return null;
+            } elseif (self::isFileUpload($filesValue)) {
+                return $filesValue;
+            } elseif (\is_array($filesValue)) {
+                return $filesValue;
+            } else { // $filesValue has a non-array value
+                return $filesValue;
+            }
```

Comment:
Can you please elaborate what are the purpose of these conditions? It looks like this block can be simplified with:
```php
if (null === $paramsValue) {
    return $filesValue;
}
```

Reply:
you're right. I expanded the matrix for readability but forgot to reoptimize. done.